### ISSUE TITLE
Implement editor for Mobile

### ISSUE THREAD

**Issue by sachben91** [hicommonwealth/commonwealth#6400]
## Description
This introduces mobile adaptive designs for the editor. The editor is used by the users to create threads and posts on Common. The prototype shows the experience of the user creating a post from an unscoped page i.e not from within a community. Part of this change is the "create" option being moved to the bottom nav which is covered in #6389 

## Project Owner
PM: @zakhap @sachben91 
Designer: @jessmart1213 

## Designs 
[Figma](https://www.figma.com/file/w6Tli9ApqBqgsZa7kB0Vk5/Mobile-ideation-Q4?type=design&node-id=573%3A62670&mode=dev)
Prototype
[Loom walkthrough](https://www.loom.com/share/dd5b763e288b40608da57c414710f85c)

## Feature Specs 
- The user can create a thread by clicking on "create" in the bottom nav and selecting "create thread". As mentioned above this behavior is covered in #6989 relating to mobile nav
- When the user clicks on create thread they are taken to a page where they can 
   - enter post title by clicking in the post title area 
   - choose a community to post to 
   - choose the topic within the community to post to. The topic selection remains inactive until user has selected the community
   - choose an address to post from. 
   - create the body of the post

<img width="450" alt="Screenshot 2024-01-18 at 3 08 14 PM" src="https://github.com/hicommonwealth/commonwealth/assets/38763242/78e86fb5-4b8f-420b-a52c-194b7866e556">

- The user is given the option for markdown formatting and to preview the post, similar to how they are provided these options in desktop
- When the user clicks the body of the post, the top of the mobile keyboard should show the formatting options available to them within the body of the post 
<img width="366" alt="Screenshot 2024-01-18 at 2 58 07 PM" src="https://github.com/hicommonwealth/commonwealth/assets/38763242/1bad339c-0c95-4056-8686-d4caeb651f2c">
- Within the formatting options, users have the ability to attach pictures from mobile as well 
- If the topic that the user selected to post to is gated, they are shown the following screen and is prevented from posting within that topic 
<img width="364" alt="Screenshot 2024-01-18 at 3 06 36 PM" src="https://github.com/hicommonwealth/commonwealth/assets/38763242/1d1fd0f7-ab67-49ce-bd29-2e605bd854fe">

**Comment by Miaplacidus** [comment]
@jnaviask @CowMuon 
Adding for further clarification that this work should follow the work currently contained in ticket #5949 (which updates the existing editor). Note that #5949 will be split into multiple tickets.

**Comment by CowMuon** [comment]
Teeing up for Sprint 6.  #S6

**Comment by rbennettcw** [comment]
Looks feasible from a dev standpoint.

**Comment by rbennettcw** [comment]
Estimating as a 5 to include implementation and mobile testing.

However, some questions about markdown/preview
- To confirm, the editor should only support markdown and not richtext?
- The preview button is a toggle or a button? What happens when it’s tapped?

Also, bear in mind that we can’t reliably detect if the keyboard is open. So we may need to keep the formatting options visible even when the keyboard is closed.

cc @sachben91

**Comment by masvelio** [comment]
@rbennettcw as `5` is ~3 days, it seems to be underestimated from my point of view. We need to basically build everything from scratch for this view, and do it wisely, because some the behaviour has to be reused eg for commenting on post. Thing that have to be done here:
- create new type of full screen, reusable page with buttons on top and sticky header
- create form with validation (community, topic, address)
- reuse existing new editor from #5949 and adjust it to mobile
- create new scrollable, editor bar with all editor icons + positioning it properly over the keyboard
- create bottom drawer for file selection (actually not sure if we are able to implement it)

Considering each of this task to be one day, we have one week of work and this is happy path in my opinion.

**Comment by masvelio** [comment]
@rbennettcw have you had a chance to take a look on this component (pic1). I have doubts that we are able to do it. Basically in desktop we have input  type `file` that open native window. On mobile the same input gives you native popup which for example on ios looks like this (pic2.). Not sure how this could look like in PWA though. 


![image](https://github.com/hicommonwealth/commonwealth/assets/14819225/b201cc9c-3860-4c18-9f89-99c9dc5d7de2)

<img width="1042" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/14819225/23e64b6e-0c2c-4010-b9dd-94e5b44be32f">

**Comment by rbennettcw** [comment]
@masvelio Agreed– File upload UI is dictated by the browser/OS and we can't customize that– it'll use the native UI

**Comment by masvelio** [comment]
@ForestMars @sachben91 
so to sum up:
- this needs higher estimation  for sure, my opinion is `8` in happy path
- we need some other design solution for file upload
